site/src/_pages/spec/3_implementation-approaches.mdx

```diff
@@ -14,19 +15,34 @@ We may wish to dictate that blocks meet certain standards before they are made a
 
 ## Rendering contexts
 
-This protocol specifies an interface between blocks and embedding applications, defining how data is referred to when it is communicated between the two, but leaves open:
+<InfoCard>
+
+We have published [the source code of an example embedding application, HASH](https://github.com/hashintel/hash).
+
+This is very much a work in progress we are using to explore and test assumptions and suggested approaches in the Block Protocol.
+
+As we develop this application further, we will test various rendering strategies in it, and provide abstractions for embedding applications to minimise the amount of work required to use blocks.
+
+We welcome feedback and suggestions - [please open or contribute to a discussion](https://github.com/blockprotocol/blockprotocol/discussions) to do so.
+
+</InfoCard>
+
+The protocol specifies an interface between blocks and embedding applications, defining how data is referred to when it is communicated between the two, but it leaves open:
 
 - the particular implementation and composition of block source code
 
 - how exactly data is transferred between an application and a block
 
 - how exactly blocks are rendered (and re-rendered) by applications
 
-There are various different implementations of embedding applications and blocks which would be compliant with the protocol but not with one another.
+There are various different implementations of embedding applications and blocks which would be compliant with the protocol, but not with one another.
 
-Embedding applications can maximise the number of blocks they can render by implementing a number of rendering strategies, which are then chosen depending on the detected or reported block implementation.
+In order to meet the ultimate goal of the Block Protocol and supporting ecosystem of any embedding application being able to render any block, if both are correctly configured, there has to be some way of applications automatically dealing with blocks requiring different approaches - by implementing a number of rendering strategies, with the appropriate one based on the detected or reported block implementation.
```

Comment:
Please can you have a stab at rewriting this sentence? Struggling to wrap my head around it, and it's very long.

Reply:
New version:

> The ultimate goal of the Block Protocol and supporting ecosystem is that any embedding application should be able to render any block, if both are correctly configured.
> 
> In order to achieve this, embedding applications will need to have different strategies for handling blocks which are implemented in different ways. 
> 
> For example:
> 
> - a block implemented as a React component would require an embedding application to render the block to the DOM using React, and pass the data and functions it expects as its "props"
> - in a block with an HTML entrypoint, appropriately sandboxed by the embedding application, the properties could be in the global scope of the sandbox (with the block assuming they are available)
> 
> There must be some way of applications identifying how each block should be rendered. This might be achieved via inference from the block's source (e.g. the extension of its entrypoint), or by an explicit property in a block's metadata file. Either approach will require a small number of known, agreed 'block implementation approaches' which can be identified and handled.